README.md

```diff
@@ -5,754 +5,360 @@
 
-class MyCounter {
-  count = Number;
-}
-
-// creating microstate from composed types
-create(MyCounter).state;
-// => { count: 0 }
-
-create(MyCounter).count.increment().state;
-// => { count: 1 }
-
-class MyModal {
-  isOpen = Boolean;
-  title = String;
-}
+Microstates is a pure JavaScript model authoring system designed to ease state management in component based applications.
 
-// you can restore a microstate from previous value
-create(MyModal, { isOpen: true, title: 'Hello World' }).state;
-// => { isOpen: true, title: 'Hello World' }
+By combining lazy execution, algebraic data types and structural sharing, we created
+a tool that provides a tiny API to describe complex data structures and provide a
+mechanism to change the value in immutable way.
 
-create(MyModal, { isOpen: true, title: 'Hello World' }).isOpen.toggle().state;
-// => { isOpen: false, title: 'Hello World' }
+## Features
 
-// lets compose multiple types together
-class MyType {
-  modal = MyModal;
-  counter = MyCounter;
-}
+With microstates added to your project, you get: 
 
-create(MyType).state;
-// => {
-//  modal: { isOpen: false, title: '' },
-//  counter: { count: 0 }
-// }
-
-create(MyType)
-  .counter.increment()
-  .modal.title.set('Hello World')
-  .modal.isOpen.set(true).state;
-// => {
-//  modal: { isOpen: true, title: 'Hello World' },
-//  counter: { count: 1 }
-// }
-```
+- 🍣 Atomic composable type system
+- 🍱 Effortless composition of types
+- 💎 Pure immutable state transitions without reducers
+- 🔭 Optional integration with Observables
+- 🎯 Transpilation free type system
+- ⚛ Use in Node.js and web applications
+- 🦋 Easiest way to express state machines
 
-## microstate constructor
+But, most imporantly, Microstates makes working with **state fun**. 
 
-Microstate constructor creates a microstate. It accepts two arguments, `Type` class that describes the structure of your state and `value` that is the initial state.
+## 🤗 Having fun with state 
 
-```js
-// initial value is undefined so state will be default value of String which is an empty string
-create(String).state;
-// => ''
+When was the last time you had fun working with state in JavaScript applications? For many, the answer is probably never because state management in JavaScript is an endless game of choosing from compromises. You can choose to go fully immutable and end up writing endless reducers. You can go mutable and everything magically becomes an observable. Or you can `setState` and loose the benefit of serialization and time travel debugging. 
 
-// initial value is 'hello world' so state will be hello world.
-create(String, 'hello world').state;
-// => 'hello world'
-```
+Unlike the view layer, where most frameworks agree on some variation of React components, state management is alloot less certain. It's less certain because none of the available tools strike the balance that React components introduced to the view layer. 
```

Comment:
typo 'alloot'

Reply:
It's an inside joke for the Frontsiders ;)